api/types/swarm/task.go

```diff
@@ -40,13 +40,14 @@ type Task struct {
 	Meta
 	Annotations
 
-	Spec                TaskSpec            `json:",omitempty"`
-	ServiceID           string              `json:",omitempty"`
-	Slot                int                 `json:",omitempty"`
-	NodeID              string              `json:",omitempty"`
-	Status              TaskStatus          `json:",omitempty"`
-	DesiredState        TaskState           `json:",omitempty"`
-	NetworksAttachments []NetworkAttachment `json:",omitempty"`
+	Spec                     TaskSpec            `json:",omitempty"`
+	ServiceID                string              `json:",omitempty"`
+	Slot                     int                 `json:",omitempty"`
+	NodeID                   string              `json:",omitempty"`
+	Status                   TaskStatus          `json:",omitempty"`
+	DesiredState             TaskState           `json:",omitempty"`
+	NetworksAttachments      []NetworkAttachment `json:",omitempty"`
+	AssignedGenericResources []*GenericResource  `json:",omitempty"`
```

Comment:
This field name is extremely awkward. Can there every be unassigned resources? These resources belong to the task and are assumed to be assigned.